### PR TITLE
Feature refactor prompts for spreadsheet

### PR DESCRIPTION
**Pull request by spikelynch**
- Configuration is now done with a spreadsheet instead of yaml
- JSON examples for the LLM are now generated automatically from the spreadsheet
- added some extra methods to make the Classifier object easier to work with
- allows overriding the cache when you want to test prompts
- better testing for prompt parsing and generation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,9 +68,14 @@ To run the `classify` command, use `poetry run`:
 poetry run classify --config config.json
 ```
 
-If you re-run the classifier, it will look in the cache for each case / prompt combination and return a cached result if it exists, rather than going to the LLM. For now, the only way to stop caching is to delete the cache file or directory for a prompt or case.
+If you re-run the classifier, it will look in the cache for each case / prompt
+combination and return a cached result if it exists, rather than going to the
+LLM. For now, the only way to stop caching is to delete the cache file or
+directory for a prompt or case.
```
Comment (xinwei89) [inline README.md:75]: Now we have `no_cache` flag, should this sentence be changed?

**Review by xinwei89 — COMMENTED** [review]
I think it's good we can directly parse the spreadsheet, might be easier for users to edit the prompts now. 

One comment is maybe we can have type hints in the definition line of functions or class methods for better readability?

**Comment by spikelynch** [comment]
> I think it's good we can directly parse the spreadsheet, might be easier for users to edit the prompts now.
> 
> One comment is maybe we can have type hints in the definition line of functions or class methods for better readability? Specifically for `classifier.py` and `prompts.py`

I started this last week but then got sick again - am leaving it on its own branch for now

File: README.md [hunk 2]
```diff
@@ -111,71 +118,84 @@ keys in the root directory of the repo.
 
 ## Prompts
 
-The following is an example of one of the prompts from `prompts_grouped.yaml`.
+Prompts are configured using an Excel spreadsheet with the following sheets:
```
[the remark this answers is not in the document]
Comment (spikelynch) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (spikelynch) [reply README.md]: Done (and made sure it's available on the command line as well

**Pull request merged**